.github/workflows/push.yml

```diff
@@ -0,0 +1,70 @@
+name: Push
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2 # checkout visx + this commit
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12'
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: 📡 Install dependencies
+        run: yarn install --frozen-lockfile --ignore-engines
+
+      - name: 🛠 Build packages
+        run: yarn build
+
+      - name: 📐 Commit package sizes
+        run: |
+          yarn build:sizes	
+          git config user.name github-actions	
+          git config user.email github-actions@github.com	
+          git add .	
+          git commit -m "build(${GITHUB_SHA}): auto-commit package sizes"	
+          git push
+
+      - name: 🚀 Build and deploy gallery
+        # below we
+        # - setup git credentials provided via actions/checkout@v2
+        # - initialize gh-pages-branch as an orphan branch so we don't build history
+        # - checkout the current commit and create gh-pages-root-dir/ as a new worktree
+        #   - outside that directory HEAD is detached at $GITHUB_SHA
+        #   - within that directory we are on the gh-pages-branch we just initialized
+        #     *worktree initialization should be in a root dir, otherwise the worktree inherits nested directories
+        # - build the static next.js site and copy the output into gh-pages-root-dir/
+        #   - we can't output directly into gh-pages-root-dir/ because next wipes the folder including .git
+        # - commit the demo site within gh-pages-root-dir/ onto the gh-pages-branch
+        # - push gh-pages-branch to visx as gh-pages. we overwrite history every time so it must be forced
+        run: |
+          git config user.name github-actions
```

Comment:
I usually use this package to handle deployment to github pages.
Would it simplify the script here?
https://www.npmjs.com/package/gh-pages

```
"scripts": {
  "deploy": "gh-pages -d dist"
}
```

```
yarn build
yarn deploy
```

Reply:
Currently we deploy to `gh-pages` with a simple script in `@visx/demo` that is similar to these commands. the complexity comes from the creation of a new git worktree without history in the action environment.

tbh I spent so much time on this already I'd rather just leave it. I tried a `gh-pages` action someone made and it worked, but it requires special handling of credentials which then causes the package size commit to fail. so would have to duplicate the `yarn && yarn build` steps multiple times which isn't ideal.